test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,27 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), publish_message_bad_arguments) {
+  test_msgs__msg__BasicTypes input_message{};
+  ASSERT_TRUE(test_msgs__msg__BasicTypes__init(&input_message));
```

Comment:
@Lobotuerk it'd seem like you're missing a `test_msgs__msg__BasicTypes__fini` call to not leak the message once the test is finished.

Reply:
I was seeing other tests that did not have the finish after initializing the basic messages

Reply:
Anyways, added

Reply:
If you have any references to tests that have success `ini` calls without `fini` calls without you can raise an issue or open a PR, because they are probably leaking memory and is not desired.

Reply:
@Blast545 https://github.com/ros2/rmw_implementation/blob/master/test_rmw_implementation/test/test_serialize_deserialize.cpp,
 maybe I'm wrong, but I see 3 inits and no fini

Reply:
That's an omission of my own! We have to fix it :)